cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -47,7 +86,28 @@ public ScheduleSpecification deserialize(JsonElement json, Type typeOfT, JsonDes
     throws JsonParseException {
     JsonObject jsonObj = json.getAsJsonObject();
 
-    Schedule schedule = context.deserialize(jsonObj.get("schedule"), Schedule.class);
+    JsonElement scheduleTypeJson = jsonObj.get("scheduleType");
+    ScheduleType scheduleType;
+    if (scheduleTypeJson == null) {
+      // For backwards compatibility with spec persisted with older versions than 2.8, we need these lines
+      scheduleType = ScheduleType.ORIGINAL_TIME;
```

Comment:
In order to maintain the backward compatibility, I think we can use the private variable `SPECIFICATION_VERSION` in the specification classes. Whenever we update the fields in these classes which might affect the serialization and deserialization, this version can be bumped up. Thoughts?